tests/ga/test_update.py

```diff
@@ -1265,79 +1266,69 @@ def test_run_latest_creates_only_one_signal_handler(self, mock_signal):
         self._test_run_latest()
         self.assertEqual(0, mock_signal.call_count)
 
-    def _test_run(self, invocations=1, calls=None, enable_updates=False, sleep_interval=(6,)):
-        if calls is None:
-            calls = [call.run()]
+    def _test_run(self, invocations=1, calls=1, enable_updates=False, sleep_interval=(6,)):
         conf.get_autoupdate_enabled = Mock(return_value=enable_updates)
 
-        # Note:
-        # - Python only allows mutations of objects to which a function has
-        #   a reference. Incrementing an integer directly changes the
-        #   reference. Incrementing an item of a list changes an item to
-        #   which the code has a reference.
-        #   See http://stackoverflow.com/questions/26408941/python-nested-functions-and-variable-scope
-        iterations = [0]
-
-        def iterator(*args, **kwargs):  # pylint: disable=unused-argument
-            iterations[0] += 1
-            if iterations[0] >= invocations:
-                self.update_handler.running = False
-            return
+        def iterator(*_, **__):
+            iterator.count += 1
+            if iterator.count <= invocations:
+                return True
+            return False
+        iterator.count = 0
 
         fileutil.write_file(conf.get_agent_pid_file_path(), ustr(42))
 
         with patch('azurelinuxagent.ga.exthandlers.get_exthandlers_handler') as mock_handler:
-            with patch('azurelinuxagent.ga.remoteaccess.get_remote_access_handler') as mock_ra_handler:
-                with patch('azurelinuxagent.ga.update.get_monitor_handler') as mock_monitor:
-                    with patch('azurelinuxagent.ga.update.get_env_handler') as mock_env:
-                        with patch('azurelinuxagent.ga.update.get_collect_logs_handler') as mock_collect_logs:
-                            with patch('azurelinuxagent.ga.update.get_send_telemetry_events_handler') as mock_telemetry_send_events:
-                                with patch('azurelinuxagent.ga.update.get_collect_telemetry_events_handler') as mock_event_collector:
-                                    with patch('azurelinuxagent.ga.update.initialize_event_logger_vminfo_common_parameters'):
-                                        with patch('azurelinuxagent.ga.update.is_log_collection_allowed', return_value=True):
-                                            with patch('time.sleep', side_effect=iterator) as sleep_mock:
```

Comment:
those tests were mocking the global sleep(), which make them extremely fragile. They broke. Now they mock is_running.

BTW - Note the amount of mocks in these tests. THIS IS NOT THE WAY TO WRITE TESTS.